stingray/crossspectrum.py

```diff
@@ -0,0 +1,380 @@
+            if lc_1 is not None or lc_2 is not None:
+                 print("You can't do a cross spectrum with just one "
+                         "light curve!")
+            # else:
+            #      print("Please specify input light curves!")
+            self.freq = None
+            self.cs = None
+            self.df = None
+            self.nphots_1 = None
+            self.nphots_2 = None
+            self.m = 1
+            self.n = None
+            return
+        self._make_crossspectrum(lc_1, lc_2)
+
+    def _make_crossspectrum(self, lc_1, lc_2):
+
+        ## make sure the inputs work!
+        assert isinstance(lc_1, lightcurve.Lightcurve), \
+                        "lc_1 must be a lightcurve.Lightcurve object!"
+        assert isinstance(lc_2, lightcurve.Lightcurve), \
+                        "lc_2 must be a lightcurve.Lightcurve object!"
+
+
+        ## total number of photons is the sum of the
+        ## counts in the light curve
+        self.nphots_1 = np.sum(lc_1.counts)
+        self.nphots_2 = np.sum(lc_2.counts)
+
+
+        ## the number of data points in the light curve
+        assert lc_1.counts.shape[0] == lc_2.counts.shape[0], \
+            "Light curves do not have same number of time bins per segment."
+        self.n = lc_1.counts.shape[0]
+
+        assert lc_1.tseg == lc_2.tseg, "Light curves do not have same tseg."
+
+        ## the frequency resolution
+        self.df = 1.0/lc_1.tseg
+
+        ## the number of averaged periodograms in the final output
+        ## This should *always* be 1 here
+        self.m = 1
+
+        ## make the actual Fourier transform and compute cross spectrum
+        self.freq, self.unnorm_cross = self._fourier_cross(lc_1, lc_2)
+
+        ## If co-spectrum is desired, normalize here. Otherwise, get raw back
+        ## with the imaginary part still intact.
+        self.cs = self._normalize_crossspectrum(self.unnorm_cross, lc_1.tseg)
+
+    def _fourier_cross(self, lc_1, lc_2):
+        """
+        Fourier transform the two light curves, then compute the cross spectrum.
+        computed as CS = lc_1 x lc_2* (where lc_2 is the one that gets
+        complex-conjugated)
+
+        Parameters
+        ----------
+        lc_1: lightcurve.Lightcurve object
+            One light curve to be Fourier transformed. Ths is the band of
+            interest or channel of interest.
+
+        lc_2: lightcurve.Lightcurve object
+            Another light curve to be Fourier transformed. This is the reference
+            band.
+
+        Returns
+        -------
+        fr: numpy.ndarray
+            The squared absolute value of the Fourier amplitudes
+
+        """
+        fourier_1 = scipy.fftpack.fft(lc_1.counts)  # do Fourier transform 1
+        fourier_2 = scipy.fftpack.fft(lc_2.counts)  # do Fourier transform 2
+
+        assert lc_1.counts.shape[0] == lc_2.counts.shape[0], \
+                "Light curves do not have the same shape."
```

Comment:
Given that we have this assertion in `_make_crossspectrum` above, do we need it here again? It might also make sense to move the assertion in line 146 into `_make_crossspectrum`, too.